seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/xugu/XuguTypeConverter.java

```diff
@@ -0,0 +1,381 @@
+    public static final long MAX_BINARY_LENGTH = POWER_2_16 - 4;
+    public static final XuguTypeConverter INSTANCE = new XuguTypeConverter();
+
+    @Override
+    public String identifier() {
+        return DatabaseIdentifier.XUGU;
+    }
+
+    @Override
+    public Column convert(BasicTypeDefine typeDefine) {
+        PhysicalColumn.PhysicalColumnBuilder builder =
+                PhysicalColumn.builder()
+                        .name(typeDefine.getName())
+                        .sourceType(typeDefine.getColumnType())
+                        .nullable(typeDefine.isNullable())
+                        .defaultValue(typeDefine.getDefaultValue())
+                        .comment(typeDefine.getComment());
+
+        String xuguDataType = typeDefine.getDataType().toUpperCase();
+        switch (xuguDataType) {
+            case XUGU_BOOLEAN:
+            case XUGU_BOOL:
+                builder.dataType(BasicType.BOOLEAN_TYPE);
+                break;
+            case XUGU_TINYINT:
+                builder.dataType(BasicType.BYTE_TYPE);
+                break;
+            case XUGU_SMALLINT:
+                builder.dataType(BasicType.SHORT_TYPE);
+                break;
+            case XUGU_INT:
+            case XUGU_INTEGER:
+                builder.dataType(BasicType.INT_TYPE);
+                break;
+            case XUGU_BIGINT:
+                builder.dataType(BasicType.LONG_TYPE);
+                break;
+            case XUGU_FLOAT:
+                builder.dataType(BasicType.FLOAT_TYPE);
+                break;
+            case XUGU_DOUBLE:
+                builder.dataType(BasicType.DOUBLE_TYPE);
+                break;
+            case XUGU_NUMBER:
+            case XUGU_DECIMAL:
+            case XUGU_NUMERIC:
+                Preconditions.checkArgument(typeDefine.getPrecision() > 0);
+
+                DecimalType decimalType;
+                if (typeDefine.getPrecision() > DEFAULT_PRECISION) {
+                    log.warn("{} will probably cause value overflow.", XUGU_DECIMAL);
+                    decimalType = new DecimalType(DEFAULT_PRECISION, DEFAULT_SCALE);
+                } else {
+                    decimalType =
+                            new DecimalType(
+                                    typeDefine.getPrecision().intValue(),
+                                    typeDefine.getScale() == null
+                                            ? 0
+                                            : typeDefine.getScale().intValue());
+                }
+                builder.dataType(decimalType);
+                builder.columnLength(Long.valueOf(decimalType.getPrecision()));
+                builder.scale(decimalType.getScale());
+                break;
+
+            case XUGU_CHAR:
+            case XUGU_NCHAR:
+            case XUGU_VARCHAR:
+            case XUGU_VARCHAR2:
+                builder.dataType(BasicType.STRING_TYPE);
+                builder.columnLength(typeDefine.getLength());
```

Comment:
this length is char length or byte length ？

Reply:
is char length

Reply:
Because multi-byte characters are handled differently in various databases, the byte length is uniformly used in seatunnel

use `TypeDefineUtils.charTo4ByteLength(your length)`

Reply:
https://github.com/apache/seatunnel/blob/dev/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/psql/PostgresTypeConverter.java#L195

https://github.com/apache/seatunnel/blob/dev/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/oracle/OracleTypeConverter.java#L169C25-L169C40

Reply:
I have handled it in XuguTypeMapper

Reply:
I have handled it in XuguTypeMapper